lib/mongodbQueue.js

```diff
@@ -72,55 +74,45 @@ MongodbQueue.prototype.create = function(cb) {
       } else {
         debug('[%s] lock %s acquired. Continue.', self.queueName, self.lockName);
         async.waterfall([
-          function createIndexes(callback) {
-            debug('[%s] creating queue indexes', self.queueName);
-            self.queue.createIndexes(callback);
-          },
-          function addExtraIdIndex(createdIndex, callback) {
-            debug('[%s] adding extra _id index for queue', self.queueName);
-            collection.createIndex({ deleted : 1, visible : 1, _id : 1}, callback);
-          },
-          function addExtraAckIndex(createdIndex, callback) {
-            debug('[%s] adding extra ack index for queue', self.queueName);
-            collection.createIndex({ ack: 1, visible : 1, deleted : 1}, callback);
-          },
+          // Check if there's already a TTL index for the 'deleted' field.
+          // If there is, and the TTL value is different, delete it so it can be
+          // recreated by the mongodb-queue module
           function listIndexes(createdIndex, callback) {
             debug('[%s] list existing indexes', self.queueName);
             collection.indexInformation({full: true}, callback);
           },
           function checkIndexTTL(indexInfo, callback) {
             debug('[%s] found queue indexInfo : %j', self.queueName, indexInfo);
             var existingIndex = _.findWhere(indexInfo, {name: indexName});
-            var needDrop = false;
-            var needCreate = false;
 
             if (!existingIndex) {
-              needCreate = true;
+              return callback(null, false);
             } else if (existingIndex.expireAfterSeconds !== self.queueTTL) {
```

Comment:
No. Good spot. this might be related to the failing tests.
This should probably be `this.queueOptions.ttl`